Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -552,14 +578,22 @@ public Declaration FindSelectedDeclaration(CodePane activeCodePane)
                                    item.DeclarationType != DeclarationType.ModuleOption &&
                                    item.DeclarationType != DeclarationType.Class &&
                                    item.DeclarationType != DeclarationType.Module &&
-                                   (IsSelectedDeclaration(selection, item) ||
-                                    item.References.Any(reference => IsSelectedReference(selection, reference))));
+                                   (IsSelectedDeclaration(selection, item) || item.References.Any(reference => reference.Declaration.Equals(item) && IsSelectedReference(selection, reference))))
+                    .ToList();
                 try
                 {
-                    var match = matches.SingleOrDefault() ?? AllUserDeclarations
```

Comment:
I haven't looked it at yet, but there's an issue for it here: https://github.com/rubberduck-vba/Rubberduck/issues/1365.